scout/server/blueprints/login/views.py

```diff
@@ -45,66 +47,37 @@ def load_user(user_id):
 
 @login_bp.route("/login", methods=["GET", "POST"])
 @public_endpoint
-def login():
+def login() -> Response:
     """Login a user if they have access."""
 
-    if current_app.config.get("USERS_ACTIVITY_LOG_PATH"):
-        if request.form.get("consent_checkbox") is None and "consent_given" not in session:
-            flash(
-                "Logging user activity is a requirement for using this site and accessing your account. Without consent to activity logging, you will not be able to log in into Scout.",
-                "warning",
-            )
-            return redirect(url_for("public.index"))
-        session["consent_given"] = True
+    if not controllers.handle_user_consent(request.form):
+        return redirect(url_for("public.index"))
 
-    user_id = None
-    user_mail = None
+    user_id: Optional[str] = None
+    user_mail: Optional[str] = None
 
     if current_app.config.get("LDAP_HOST", current_app.config.get("LDAP_SERVER")):
-        ldap_authorized = controllers.ldap_authorized(
-            request.form.get("ldap_user"), request.form.get("ldap_password")
-        )
-        if ldap_authorized is True:
-            user_id = request.form.get("ldap_user")
-        else:
-            flash("User not authorized by LDAP server", "warning")
+        user_id = controllers.ldap_login(request.form)
+        if user_id is None:
             return redirect(url_for("public.index"))
 
     elif current_app.config.get("GOOGLE"):
         if session.get("email"):
             user_mail = session["email"]
-            session.pop("email", None)
         else:
-            redirect_uri = url_for(".authorized", _external=True)
-            try:
-                return oauth_client.google.authorize_redirect(redirect_uri)
-            except Exception as ex:
-                flash("An error has occurred while logging in user using Google OAuth")
-
-    elif request.form.get("email"):  # log in against Scout database
-        user_mail = request.form["email"]
-        LOG.info("Validating user %s email %s against Scout database", user_id, user_mail)
-
-    user_obj = store.user(email=user_mail, user_id=user_id)
-    if user_obj is None:
-        flash("User not found in Scout database", "warning")
-        return redirect(url_for("public.index"))
+            # Redirect to Google OAuth if not completed
+            return controllers.google_login()
 
-    user_obj["accessed_at"] = datetime.now()
-    if session.get("name"):  # These args come from google auth
-        user_obj["name"] = session.get("name")
-        user_obj["locale"] = session.get("locale")
-    store.update_user(user_obj)
+    elif request.form.get("email"):
+        user_mail = controllers.database_login(request.form)
 
-    user_dict = LoginUser(user_obj)
-    return perform_login(user_dict)
+    return controllers.validate_and_login_user(user_mail=user_mail, user_id=user_id)
 
 
 @login_bp.route("/authorized")
 @public_endpoint
 def authorized():
```

Comment:
I'd line to rename this function to something like `google_callback` or `google_authorize`.  But to do so we need to edit the settings in the Google Admin console and register the new callback endpoint. 

We could perhaps add another uri in these settings:

<img width="576" alt="image" src="https://github.com/user-attachments/assets/f7cb2a0d-d8c2-4042-9cc1-2454aa9df53b" />

But then if somebody else apart from us is using google for logging in then their login would be broken. Perhaps one could use some alias or something..

Reply:
Mm, I think they are going to be picky about aliases for these in particular, but I agree about the naming. I would enjoy something like `oath2_callback`. We could do this for the next major? 😁